File: problems/226. Invert Binary Tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    
    // Recursive
    public TreeNode invertTree(TreeNode root) {
        swap(root);
        return root;
        
    }
    
    public void swap(TreeNode node) {
        if (node == null)  return;
        TreeNode temp = node.left;
        node.left = node.right;
        node.right = temp;
        swap(node.left);
        swap(node.right);
    }
    
    // Iterative
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        queue.add(root);
        
        while (!queue.isEmpty()) {
            TreeNode curr = queue.poll();
            TreeNode temp = curr.left;
            curr.left = curr.right;
            curr.right = temp;
            
            if (curr.right != null) queue.add(curr.right);
            if (curr.left != null) queue.add(curr.left);
        }
        
        return root;
    }
}

File: problems/226. Invert Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from collections import deque as queue


class Solution:
    # Recursive
    def invertTree(self, root: TreeNode) -> TreeNode:
        self.swap(root)
        return root

    def swap(self, node):
        if not node:
            return

        node.left, node.right = node.right, node.left
        self.swap(node.right)
        self.swap(node.left)

    # Iterative

    def invertTree(self, root):
        if not root:
            return None
        q = queue([root])
        while q:
            curr = q.popleft()
            curr.left, curr.right = curr.right, curr.left
            if curr.left:
                q.append(curr.left)
            if curr.right:
                q.append(curr.right)
        return root
